### PR TITLE
entity things

### PR DESCRIPTION
**Pull request by eristhea**
~~knocking out some unmapped entity-related classes~~. never mind. this became a 100% of entity AI

refactors:
`run` -> `create`: these methods create a TaskControl, not run a task
`ArmadilloWhatever` -> `MobSensor`: the mojang name is `MobSensor`, which indicates that it will in the future be used for things other than armadillos. it's also possible for modders to use this to sense any entity. best to rename now!

### CODE REVIEW

File: mappings/net/minecraft/entity/ai/brain/task/StrollTask.mapping
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/ai/brain/task/StrollTask.mapping:1]: `TravelTask`?
`IdleTravelTask`?
`MeanderTask`?

Trying to come up with something that's like stroll but not necessarily on foot...

File: mappings/net/minecraft/entity/ai/brain/task/TaskBuilder.mapping [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			ARG 3 time
+		METHOD m_ioeqqkda toDebugString ()Ljava/lang/String;
+	CLASS C_hzjatjhv Constant
+		METHOD <init> (Ljava/lang/Object;)V
+			ARG 1 value
+		METHOD <init> (Ljava/lang/Object;Ljava/util/function/Supplier;)V
+			ARG 1 value
+			ARG 2 debugStringSupplier
+	CLASS C_loqvwuht Instance
+		METHOD ap2 (Lcom/mojang/datafixers/kinds/App;Lcom/mojang/datafixers/kinds/App;Lcom/mojang/datafixers/kinds/App;)Lcom/mojang/datafixers/kinds/App;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/entity/ai/brain/task/TaskBuilder.mapping:45]:
```suggestion
		METHOD app2 (Lcom/mojang/datafixers/kinds/App;Lcom/mojang/datafixers/kinds/App;Lcom/mojang/datafixers/kinds/App;)Lcom/mojang/datafixers/kinds/App;
```
?

File: mappings/net/minecraft/server/world/ServerWorld.mapping [hunk 2]
```diff
@@ -117,10 +121,13 @@ CLASS net/minecraft/unmapped/C_bdwnwhiu net/minecraft/server/world/ServerWorld
 	METHOD m_jhwkhqgo isNearOccupiedPointOfInterest (Lnet/minecraft/unmapped/C_hynzadkk;I)Z
 		ARG 1 pos
 		ARG 2 maxDistance
+	METHOD m_jyqkjiyo areEntitiesLoadedAndTicking (J)Z
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/server/world/ServerWorld.mapping]:
```suggestion
	METHOD m_jyqkjiyo areChunkEntitiesLoadedAndTicking (J)Z
```
To me the `pos` param is confusing without `chunk` (pos of what?)

File: mappings/net/minecraft/entity/ai/brain/task/TaskBuilder.mapping (hunk 1 above)
Comment (eristhea) [reply mappings/net/minecraft/entity/ai/brain/task/TaskBuilder.mapping:45]: I must have marked an automapped name as deobf, that name is included in the jar for some reason

File: mappings/net/minecraft/entity/ai/brain/task/StrollTask.mapping
Comment (eristhea) [reply mappings/net/minecraft/entity/ai/brain/task/StrollTask.mapping:1]: I'd have to look deeper into the usages to choose one, but stroll is definitely not ideal

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/entity/ai/brain/task/StrollTask.mapping:1]: mojang calls it `RandomStroll`, and from a glance it's often an idle task and sometimes paired with a waiting task
So I'm leaning `MeanderTask`

Comment (eristhea) [reply mappings/net/minecraft/entity/ai/brain/task/StrollTask.mapping:1]: I'll put that change in the PR!

**Review by supersaiyansubtlety — APPROVED** [review]

File: mappings/net/minecraft/entity/ai/brain/task/MeleeAttackTask.mapping [hunk 3]
```diff
@@ -1,3 +1,8 @@
 CLASS net/minecraft/unmapped/C_gpxkysqi net/minecraft/entity/ai/brain/task/MeleeAttackTask
+	METHOD m_bsseqsar create (I)Lnet/minecraft/unmapped/C_yfpegcpm;
+		ARG 0 cooldown
 	METHOD m_njoxpzqb isHoldingUsableRangedWeapon (Lnet/minecraft/unmapped/C_dxkfswlz;)Z
+		ARG 0 entity
 		ARG 1 entity
```
Comment (IotaBread) [inline mappings/net/minecraft/entity/ai/brain/task/MeleeAttackTask.mapping]: leftover?

File: mappings/net/minecraft/entity/ai/brain/task/WorkStationCompetitionTask.mapping [hunk 4]
```diff
@@ -1,8 +1,15 @@
...
 	METHOD m_wewjuzkg isUsingWorkStationAt (Lnet/minecraft/unmapped/C_ovcqqyqp;Lnet/minecraft/unmapped/C_cjzoxshv;Lnet/minecraft/unmapped/C_pdtkdbte;)Z
-		ARG 1 pos
+		ARG 0 pos
+		ARG 1 poiType
+		ARG 2 villager
 		ARG 3 villager
 	METHOD m_xmcctqkk isCompletedWorkStation (Lnet/minecraft/unmapped/C_cjzoxshv;Lnet/minecraft/unmapped/C_yhqxfbuc;)Z
+		ARG 0 poiType
+		ARG 1 profession
 		ARG 2 profession
```
Comment (IotaBread) [inline mappings/net/minecraft/entity/ai/brain/task/WorkStationCompetitionTask.mapping]: leftover again

**Review by IotaBread — APPROVED** [review]

**Comment by eristhea** [comment]
this is why we need to make enigma drop invalid parameters :p
I'll remove them!

**Comment by eristhea** [comment]
confirmed that both parameters are out of bounds and removed

**Pull request merged**